### PR TITLE
Added role and user repositories

### PR DESCRIPTION
**Pull request by Bozhokmaria**
…es/services

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/role/repository/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.exadel.discountwebapp.role.repository;
+
+import com.exadel.discountwebapp.role.entity.Role;
+import org.springframework.data.repository.CrudRepository;
+
+public interface RoleRepository extends CrudRepository<Role, Long> {
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/role/repository/RoleRepository.java:8]: add @Repository annotation

File: src/main/java/com/exadel/discountwebapp/user/repository/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.exadel.discountwebapp.user.repository;
+
+import com.exadel.discountwebapp.user.entity.User;
+import org.springframework.data.repository.CrudRepository;
+
+public interface UserRepository extends CrudRepository<User, Long> {
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/repository/UserRepository.java:8]: add @Repository annotation

Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/repository/UserRepository.java:8]: done

File: src/main/java/com/exadel/discountwebapp/role/repository/RoleRepository.java (hunk 1 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/role/repository/RoleRepository.java:8]: done

**Review by ahlukhouski2 — APPROVED** [review]

**Review by VolodymyrBushko — APPROVED** [review]

File: src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+public class UserRequestVO extends UserBaseVO {
+
+    public static User toUser(UserRequestVO request){
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java]: move to UserMapper

File: src/main/java/com/exadel/discountwebapp/user/vo/UserResponseVO.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+@Getter
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+public class UserResponseVO extends UserBaseVO {
+
+    private long id;
+
+    public static UserResponseVO fromUser(User user) {
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/vo/UserResponseVO.java]: move to UserMapper

File: src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.exadel.discountwebapp.user.vo;
+
+import com.exadel.discountwebapp.user.entity.User;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java]: use @Data instead

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public UserResponseVO findById(long id) {
+        User user = userRepository.findById(id).orElse(null);
+        return user!=null ? UserResponseVO.fromUser(user) : null;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/service/UserService.java]: formatting

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public List<UserResponseVO> findAll() {
+        List<UserResponseVO> response = new ArrayList<>();
+        userRepository.findAll()
+                        .forEach(user -> response
+                                .add(UserResponseVO.fromUser(user)));
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/service/UserService.java]: formatting

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.exadel.discountwebapp.user.entity.User;
+import com.exadel.discountwebapp.user.repository.UserRepository;
+import com.exadel.discountwebapp.user.vo.UserResponseVO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/service/UserService.java:13]: redundant empty line

File: src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+@NoArgsConstructor
+public class UserRequestVO extends UserBaseVO {
+
+    public static User toUser(UserRequestVO request){
+    return User.builder()
+    .firstName(request.getFirstName())
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java]: formatting

File: src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java (hunk 3 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java]: done

File: src/main/java/com/exadel/discountwebapp/user/vo/UserResponseVO.java (hunk 4 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/vo/UserResponseVO.java]: done

File: src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java (hunk 5 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java]: added

File: src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java (hunk 9 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/vo/UserRequestVO.java]: +

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java (hunk 7 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/service/UserService.java]: done

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java (hunk 6 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/service/UserService.java]: +

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java (hunk 8 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/service/UserService.java:13]: deleted

File: src/main/java/com/exadel/discountwebapp/user/service/UserService.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public UserResponseVO findById(long id) throws UserNotFoundException {
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/service/UserService.java]: Declaring throws UserNotFoundException is redundant here

Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/service/UserService.java]: deleted

**Pull request merged**